platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java

```diff
@@ -253,6 +268,50 @@ public void getPostLogoutRequest() throws Exception {
         response.getEntity().toString().contains(postLogoutUrl));
   }
 
+  @Test
+  public void soapLogoutRequestIssueInstantOld() throws Exception {
+    HttpSessionInvalidator httpSessionInvalidator = mock(HttpSessionInvalidator.class);
+    logoutRequestService.setHttpSessionInvalidator(httpSessionInvalidator);
+
+    InputStream requestStream =
+        LogoutRequestServiceTest.class.getResourceAsStream("/SAMLSoapLogoutRequest-good.xml");
+    Response response = logoutRequestService.soapLogoutRequest(requestStream, null);
+    assertThat(response.getStatus(), is(Status.INTERNAL_SERVER_ERROR.getStatusCode()));
+    verify(httpSessionInvalidator, times(0)).invalidateSession(anyString(), any());
+  }
+
+  @Test
+  public void soapLogoutRequestBadSignature() throws Exception {
+    HttpSessionInvalidator httpSessionInvalidator = mock(HttpSessionInvalidator.class);
+    logoutRequestService.setHttpSessionInvalidator(httpSessionInvalidator);
+
+    LogoutResponse logoutResponse = mock(LogoutResponse.class);
+    doReturn(logoutResponse)
+        .when(logoutMessage)
+        .buildLogoutResponse(anyString(), anyString(), anyString());
+    doThrow(SignatureException.class)
+        .when(simpleSign)
+        .validateSignature(any(Signature.class), any(Document.class));
+
+    InputStream requestStream =
+        LogoutRequestServiceTest.class.getResourceAsStream("/SAMLSoapLogoutRequest-good.xml");
+    Response response = logoutRequestService.soapLogoutRequest(requestStream, null);
+    assertThat(response.getStatus(), is(Status.OK.getStatusCode()));
```

Comment:
❓ Should we really expect a `200` if we're throwing a signature exception?

Reply:
From section 3.2.3.3 of the [SAML Bindings v2.0 Errata Composite](https://www.oasis-open.org/committees/download.php/56780/sstc-saml-bindings-errata-2.0-wd-06-diff.pdf)

> In the case of a SAML processing error, the SOAP HTTP server [E19]SHOULD~MUST~ respond with "200 OK" and include a SAML-specified `<samlp:Status>` element in the SAML response within the SOAP body. Note that the `<samlp:Status>` element does not appear by itself in the SOAP body, but only within a SAML response of some sort.